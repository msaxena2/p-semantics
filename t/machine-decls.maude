in "resources/test-machines.maude" .

mod P-STATEMENT-TEST
  is including TEST-MACHINES .

  op initCfg : -> Cell .

  eq initCfg = <k> m1 m2 </k> <machines> .Map </machines> .

endm

search initCfg =>! <k> .Stmt </k> <machines> M1 |-> S1:Stmt M2 |-> S2:Stmt </machines>
  such that S1:Stmt == m1 /\ S2:Stmt == m2 .

quit .
