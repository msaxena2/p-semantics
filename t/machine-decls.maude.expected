==========================================
search in P-STATEMENT-TEST : initCfg =>! 
<k>  .Stmt
</k> 
<machines>  
    M1 |-> S1:Stmt   
    M2 |-> S2:Stmt  
</machines> such that S1:Stmt == m1 = true /\ S2:Stmt == m2 = true .

Solution 1 (state 2)
S1:Stmt --> machine M1 ; {
  x : int ;
  y : int ;
}
S2:Stmt --> machine M2 ; {
  a : int ;
  b : int ;
}

No more solutions.
Bye.
